From: Ben Hutchings <ben@decadent.org.uk>
Subject: hid: Avoid ABI change in 3.2.31

Commit b6787242f32700377d3da3b8d788ab3928bab849 ('HID: hidraw: add
proper error handling to raw event reporting') changed the return type
of hid_report_raw_event() and hidraw_report_event() from void to int
(an error number).  Any existing OOT callers are going to ignore this
value, whether or not they get recompiled.  Therefore, hide the change
from genksyms.
--- a/include/linux/hid.h
+++ b/include/linux/hid.h
@@ -875,8 +875,14 @@ static inline int hid_hw_power(struct hi
 	return hdev->ll_driver->power ? hdev->ll_driver->power(hdev, level) : 0;
 }
 
+#ifdef __GENKSYMS__
+/* Old callers will ignore the return value even if we change the return type */
+void hid_report_raw_event(struct hid_device *hid, int type, u8 *data, int size,
+		int interrupt);
+#else
 int hid_report_raw_event(struct hid_device *hid, int type, u8 *data, int size,
 		int interrupt);
+#endif
 
 extern int hid_generic_init(void);
 extern void hid_generic_exit(void);
--- a/include/linux/hidraw.h
+++ b/include/linux/hidraw.h
@@ -76,7 +76,12 @@ struct hidraw_list {
 #ifdef CONFIG_HIDRAW
 int hidraw_init(void);
 void hidraw_exit(void);
+#ifdef __GENKSYMS__
+/* Old callers will ignore the return value even if we change the return type */
+void hidraw_report_event(struct hid_device *, u8 *, int);
+#else
 int hidraw_report_event(struct hid_device *, u8 *, int);
+#endif
 int hidraw_connect(struct hid_device *);
 void hidraw_disconnect(struct hid_device *);
 #else
